stats.py

```diff
@@ -1,17 +1,70 @@
 """Some simple statistical functions."""
 
-def mean(data):
-    """Takes the mean/average of a list of numbers, a simple measure of 'middle'. See: https://en.wikipedia.org/wiki/Arithmetic_mean"""
-    pass
+import numbers
+import collections
+import math
 
-def median(numbers):
-    """Finds the median or middle of a list of sorted numbers, a simple measure of 'middle'. See: https://en.wikipedia.org/wiki/Median"""
-    pass
+def mean(number_list):
+    """Takes the mean/average of a list of numbers, a simple measure of 'middle'.
+    See: https://en.wikipedia.org/wiki/Arithmetic_mean"""
+    #preconditions
+    assert isinstance(number_list, collections.Iterable), "Must be iterable."
+    assert all(isinstance(x, numbers.Number) for x in number_list), "Must be numbers."
+    #function
+    answer = (sum(number_list) / len(number_list))
+    #postconditions
+    assert isinstance(answer, numbers.Number), "Mean will return a number."
+    assert answer >= min(number_list), "Mean should be higher or equal lowest number in list."
+    assert answer <= max(number_list), "Mean should be lower or equal highest number in list."
+    return answer
 
-def range(numbers):
-    """Finds ths range of a list of numbers, a simple measure of variance. See: https://www.mathsisfun.com/definitions/range-statistics-.html"""
-    pass
+def median(number_list):
+    """Finds the median or middle of a list of sorted numbers, a simple measure of 'middle'.
+    See: https://en.wikipedia.org/wiki/Median"""
+    #preconditions
+    assert isinstance(number_list, collections.Iterable), "Must be iterable."
+    assert all(isinstance(x, numbers.Number) for x in number_list), "Must be numbers."
+    assert sorted(number_list) == number_list, "Numbers must be in order smallest to largest."
+    #function
+    srtd = sorted(number_list)
```

Comment:
the number_list is already sorted. you do not need this line.

you assert on line 27 that the number list is sorted. calling sorted on them again is a waste.